.github/workflows/bazel_test.yml

```diff
@@ -0,0 +1,19 @@
+name: Bazel Test
+
+on: [push, pull_request]
```

Comment:
Final question -- what does pull_request do in addition to push? I see two checks, and I'm guessing they might be redundant?

Reply:
Good point! I guess no pull_request can skip the push step?

Reply:
Would that mean the Workflow would run after the PR is approved and before it's pushed? I guess we want that to run at PR creation time too.

Reply:
From https://help.github.com/en/actions/reference/events-that-trigger-workflows, it seems that the "push" event happens every time a commit is pushed to a repository branch. So if someone forks this repo and sends a PR from its clone, there won't be a local branch and the trigger might not run. Should we use `pull_request` instead?

Reply:
I added `pull_request` back, is it necessary to delete `push`?

Reply:
It looks like the `push` can be deleted, unless you'd like to run the check for outstanding work happening in a branch before a PR is created.

Reply:
I think the `push` is needed when people is developing on the feature branch/non-master branch, which means the `pull request` may not be created thus the tests won't be run if no `push` here.

Reply:
So either way (keep both or just pull_request) is fine with me. Leaving the choice up to you 😄